### ISSUE TITLE
[java] UnnecessaryCaseChange can not detect the case like: foo.equals(bar.toLowerCase())

### ISSUE THREAD

**Issue by wuchiuwong** [pmd/pmd#2531]
**Affects PMD Version:** 
6.22.0

**Rule:**
[UnnecessaryCaseChange](https://pmd.github.io/latest/pmd_rules_java_errorprone.html#unnecessarycasechange)

**Description:**
UnnecessaryCaseChange can not detect the case like: <code>foo.equals(bar.toLowerCase())</code>.
This rule is implemented in File: pmd-java-6.22.0-sources.jar!\net\sourceforge\pmd\lang\java\rule\errorprone\UnnecessaryCaseChangeRule.java.
Firstly, use <code>if (image == null || !(image.endsWith("toUpperCase") || image.endsWith("toLowerCase")))</code> to check if PrimarySuffix has <code>toUpperCase</code> or <code>toLowerCase</code>.
Then, use <code>if (suffix.getImage() == null || !(suffix.hasImageEqualTo("equals") || suffix.hasImageEqualTo("equalsIgnoreCase"))) {</code> to check if PrimarySuffix has <code>equals</code>.
So this rule can not detect the case：<code>foo.equals(bar.toLowerCase())</code>.

**Code Sample demonstrating the issue:**
```
foo.equals(bar.toLowerCase())
```

**Expected outcome:**

false-negative

**Running PMD through:**
CLI
